File: src/main/java/um/haberes/rest/model/CursoCargo.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "cursoId", "anho", "mes", "cargoTipoId", "legajoId" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class CursoCargo extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 3440583907430980457L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cursoCargoId = null;

	private Long cursoId = null;
	private Integer anho = 0;
	private Integer mes = 0;
	private Integer cargoTipoId = null;
	private Long legajoId = null;
	private BigDecimal horasSemanales = BigDecimal.ZERO;
	private BigDecimal horasTotales = BigDecimal.ZERO;
	private Integer designacionTipoId = null;
	private Integer categoriaId = null;
	private Byte desarraigo = 0;
	private Long cursoCargoNovedadId = null;

	@OneToOne
	@JoinColumn(name = "cursoId", insertable = false, updatable = false)
	private Curso curso;

	@OneToOne
	@JoinColumn(name = "cargoTipoId", insertable = false, updatable = false)
	private CargoTipo cargoTipo;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "designacionTipoId", insertable = false, updatable = false)
	private DesignacionTipo designacionTipo;

	@OneToOne
	@JoinColumn(name = "categoriaId", insertable = false, updatable = false)
	private Categoria categoria;

}

File: src/main/java/um/haberes/core/kotlin/model/CursoCargo.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["cursoId", "anho", "mes", "cargoTipoId", "legajoId"])])
data class CursoCargo(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cursoCargoId: Long? = null,

    var cursoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var cargoTipoId: Int? = null,
    var legajoId: Long? = null,
    var horasSemanales: BigDecimal = BigDecimal.ZERO,
    var horasTotales: BigDecimal = BigDecimal.ZERO,
    var designacionTipoId: Int? = null,
    var categoriaId: Int? = null,
    var desarraigo: Byte = 0,
    var cursoCargoNovedadId: Long? = null,

    @OneToOne
    @JoinColumn(name = "cursoId", insertable = false, updatable = false)
    var curso: Curso? = null,

    @OneToOne
    @JoinColumn(name = "cargoTipoId", insertable = false, updatable = false)
    var cargoTipo: CargoTipo? = null,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "designacionTipoId", insertable = false, updatable = false)
    var designacionTipo: DesignacionTipo? = null,

    @OneToOne
    @JoinColumn(name = "categoriaId", insertable = false, updatable = false)
    var categoria: Categoria? = null

) : Auditable()
